### DISCUSSION TITLE
Zigbee Sensor Address Changes After Power Loss

### DISCUSSION THREAD

**Discussion by zizebra** [arendst/Tasmota#21197] [Q&A] [answered]
Hi All 

I have asked about this before and could not find enough information to fully prove issue. Yesterday i was fiddling with my power cables and unfortunately ended up switching off power supply on the Sonoff Zigbee device. Upon restoration of power one of my sensors stopped reporting and registreed with a new ID. The new ID appears to be paired already and provides its status changes. I have a rule which polls any battery based sensor to read the battery level. Now what is not making sense is that whenever the device reports and the rule fires, its now reporting "missing endpoint". So the new ID cannot be polled. 

10:47:01.558 MQT: tele/zbb_9EEC7F/EB16_1/SENSOR = {"ZbReceived":{"0xEB16":{"Device":"0xEB16","0500<00":"010000010000","ZoneStatusChange":1,"ZoneStatusChangeZone":1,"Endpoint":1,"LinkQuality":97}}}
10:47:01.571 RUL: ZBRECEIVED#?#ZONESTATUSCHANGE performs "zbsend {"Device":"0xEB16","Read":{"BatteryPercentage":true}}"
10:47:01.578 MQT: stat/zbb_9EEC7F/RESULT = {"ZbSend":"Missing endpoint"}
10:48:01.525 MQT: tele/zbb_9EEC7F/EB16_1/SENSOR = {"ZbReceived":{"0xEB16":{"Device":"0xEB16","0500<00":"000000010000","ZoneStatusChange":0,"ZoneStatusChangeZone":1,"Endpoint":1,"LinkQuality":97}}}
10:48:01.535 RUL: ZBRECEIVED#?#ZONESTATUSCHANGE performs "zbsend {"Device":"0xEB16","Read":{"BatteryPercentage":true}}"
10:48:01.540 MQT: stat/zbb_9EEC7F/RESULT = {"ZbSend":"Missing endpoint"}

How does that happen?

**Comment by sfromis** [comment]
While I can't guess what happened, I'd recommend to *never* use that "short address" for anything at all, but instead assign a `ZbName` for the `IEEEAddr`. Also is much more readable, especially if you also set `ZbNameKey` on.

I also like to use `ZbNoPrefix` to avoid having that "ZbReceived" all the time, makes payloads basically the same structure as sensor payloads.

**Comment by s-hadinger** [comment]
It's always device specific. When an end-device loses connection with the coordinator, it may either simply reconnect, or silently re-pair which causes a change in short address. We have no control on this.

As mentioned by @sfromis when using `ZbName`, the short address is automatically updated when the device re-pairs based on `IEEEAddr`

**Comment by zizebra** [comment]
> It's always device specific. When an end-device loses connection with the coordinator, it may either simply reconnect, or silently re-pair which causes a change in short address. We have no control on this.
> 
> As mentioned by @sfromis when using `ZbName`, the short address is automatically updated when the device re-pairs based on `IEEEAddr`

That is precisely where my problem is. Iam using names, this device had the name zb_motion_1 and after it re-paired the existing name remained alongside the new ID.  Also note even from my previous post the IEEEAddr is incorrect please result of zbinfo. I removed the old short address ID 0x7200 and used the new short address id now being reflected 0xEB16. I now get that the short address changes but i am at a loss on the IEEEAddr which seems to be culprit.

tele/zbb_9EEC7F/zb_motion_1/SENSOR = {"zb_motion_1":{"Device":"0xEB16","Name":"zb_motion_1","IEEEAddr":"0x0000000000000000","Endpoints":[],"Config":[],"Reachable":true,"BatteryPercentage":100,"LastSeen":183,"LastSeenEpoch":1713347006,"LinkQuality":92}}
11:46:29.984 MQT: stat/zbb_9EEC7F/RESULT = {"ZbInfo":"Done"}

Zbnamekey is already on and the ZbnoPrefix, i wasnt aware. I will use this. Thank you for pointing it out. I previously reset hub hoping this would resolve seems invalid IEEEAddr but seems this does not go away. The only way is for me to remove and re-pair the device.

**Reply by sfromis** [reply]
What I meant was to never use those short addresses for **anything at all**, but **only** ZbName with the IEEEAddr. What s-hadinger suggests is that the name should automatically adapt when seeing the same IEEEAddr again

**But** when you now post this as being zeroes, it is a quite different problem from "just" not getting the same again.....

**Comment by zizebra** [comment]
> What I meant was to never use those short addresses for **anything at all**, but **only** ZbName with the IEEEAddr. What s-hadinger suggests is that the name should automatically adapt when seeing the same IEEEAddr again
> 
> **But** when you now post this as being zeroes, it is a quite different problem from "just" not getting the same again.....

thanks , so my issue with invalid IEEEAddr is unique to me only and cannot be resolved. Missinng endpoint is probably linked to the IEEEAddr

**Reply by sfromis** [reply]
IIRC, you're not the first having trouble with IEEEAddr being zeroes, but it is not an issue I know much about.

**Reply by zizebra** [reply]
ah, thanks. i thought i was the only one.

**Reply by s-hadinger** [reply, marked as answer]
IEEEaddr is announced by the device when it connects but sometimes they don't announce or the message is lost. Try pairing again, it usually solves it.
